### ISSUE TITLE
Exploit crowdfunding to raise money

### ISSUE THREAD

**Issue by freakboy3742** [pybee/paying-the-piper#3]
Kickstarter/IndieGoGo exist, and have been used to fund some open source development:
- [Django Schema Migrations](https://www.kickstarter.com/projects/andrewgodwin/schema-migrations-for-django)
- [Django PostgreSQL support](https://www.kickstarter.com/projects/mjtamlyn/improved-postgresql-support-in-django)
- [Django REST framework 3](https://www.kickstarter.com/projects/tomchristie/django-rest-framework-3)
- [Multiple template engines](https://www.indiegogo.com/projects/multiple-template-engines-for-django)

**Comment by mjtamlyn** [comment]
One of my biggest concerns with the crowdfunding model (and the DSF scholarship can be included here) is that a pretty decent chunk of the money comes from developer's own pockets. I know a lot of people who donated £100 of their own money but couldn't persuade their profitable company to do the same.

**Comment by thbar** [comment]
My own concern with this is two-fold: first this isn't recurring revenue (unlike Sidekiq Pro offering), second the type of money shown here is below what a part-time good developer can earn if he earns his living using another technique.

The lack of recurring revenue is an issue because the developer will still have to carefully manage "where his next income is going to come from", so this is not a long term view.

Currently as both a SaaS bootstrapper but also open-source developer seeking to find a recurring revenue model for one of my gems (mostly to just make sure I can provide quality time to the gem for a long time), I already thought about this but left this possibility due to above reasons.

This isn't to say nobody will be successful doing this, but yet I wanted to underline this!

**Comment by freakboy3742** [comment]
@thbar Agreed that being non-recurring is an issue. It might be better to think of Kickstarter as "Venture Capital" money, and something like Patreon as "maintenance" money. You go to Kickstarter to develop an idea; you transition to Patreon for long term maintenance once the project is established. 

The complication here is that there isn't a good transition from model A to B - so having developed an "investor" base, you need to re-sell them on the idea of ongoing expenditure for maintenance.

**Comment by ximbled** [comment]
[@TryGhost](https://www.kickstarter.com/projects/johnonolan/ghost-just-a-blogging-platform)

Raised over the £100,000 they were after to build a new blogging platform and built a non-profit around it that offers hosted solutions for low cost.

It's doable but you need to have the technical insight to build the product, your average/run-of-the-mill/great/excellent jQuery plugin isn't worth jack til someone uses it.

**Comment by sethmcleod** [comment]
On this note, [Bounty Source](https://www.bountysource.com/) seems like a great tool for fundraising and creating/collecting bug bounties.

**Comment by andrewgodwin** [comment]
My view on croundfunding is that you have to offer a very substantial thing that people already have a marked interest in - all the ones you mention above have either come from a proven historical need (DRF, Migrations) or are clearly building a well-defined new feature (Postgres, Templates).

However, I don't think this meshes entirely well with general open source development; not only is it just a one-time payment, but the idea of rewards often doesn't match well and it requires something you can entirely bound and describe up front - which is great for some projects, but less great for others (how would you describe a kickstarter to develop Python, for example? It would have to be to add a specific feature to Python).

The Patreon model is interesting, even if they do have apparently awful security (hopefully that's improving), and I think it's a model that's more worth pursuing, but the value-add needs to be changed. Artists/writers/etc on Patreon release things just for their backers first, and then for the general public shortly after; I don't think that works for open-source, but I do think a similar value-add would be required.

In particular, I think it's a good idea to look at the different people who would want to use the software and see what they might want from it:
- Single, independent developers: Exclusive forum/list to discuss issues with other backers?
- Small companies: Access to more dedicated support; perhaps a 48-hour-or-less turnaround type?
- Big companies: Direct input into feature direction (but not control!), better support again
- Enterprises: Perhaps ability to fly a developer out to be on site for a bit if they have big issues?

(not sure these are the right things, just trying to throw out examples)

I think it would be educational to look at how different markets judge buying commercial software/supplies/services, and see how we can appeal to the same need. Just relying on people's good will isn't going to work, and we'll end up disproportionately appealing to independent developers or developers on a personal level and that's not as sustainable I don't think.

Finally, I don't think it's a bad thing to massively deprioritise people who are not financially supporting a project. There still needs to be a community of people around something, of course, and I'd never advocate just listening to those who are paying you money, but I think there are nice ways of keeping a community feeling (and ultimately using that as a source of new maintainers, of course!) while giving supporters more say. People who work on the project (developers, designers, writers, triagers, moderators etc) would be given similar status to if they had contributed money, but being careful that one pull request does not immediately get you support forever.

The dark path here of course is a giant company paying loads and then demanding certain features, but if by doing that they've basically had to pay enough for a whole new core developer, surely that's a net win?

**Comment by nanuxbe** [comment]
+1 with @andrewgodwin 's version of this proposal

**Comment by aaugustin** [comment]
Since I've been asked to contribute my experience, I don't think it's a very good model because:
- few people can use it successfully: years of working for free to earn trust from the community are a prerequisite
- the amount of money it brings, while non negligible, is still small compared to the effort needed to maintain an open-source project that's sufficiently interesting to use crowdfunding
- it isn't recurring

**Comment by ximbled** [comment]
@aaugustin Which of you're statements is the one that applies with @TryGhost

Yes the founder was well respected in the community but from people that I know who have worked with him this was rightly the case, developers do recognise talent and it's not difficult to spot.

You don't have to work for free to get trust from the community. You just have to work well.

The raise was well priced and clearly structured and everyone knew they were starting a non-profit governance board from the out.

It is recurring through now traditional SaaS. There will never be a day when the apprentice at `Sallys Dog Grooming Saloon` will be able to hand build a blogging platform. You could teach every kid everywhere to write HTML from the age of 7 and if they wanted to be a hairdresser they'd still go into hair dressing.

**Comment by DanielKehoe** [comment]
I recently raised $8000 on Kickstarter to add a new feature to my [Rails Composer](http://www.railscomposer.com) project (see the [Kickstarter for Rails Composer](http://kck.st/1KzZXVC)). I'm grateful for the support but it's not sustainable. It took a month of dedicated effort to launch and promote the Kickstarter campaign. And I raised enough money to cover only one month of my expenses working full-time on the project. I can't go back next month and ask supporters to contribute again.

The campaign was successful because I already had email lists of several thousand Rails developers. If you don't have large email lists, it's difficult to successfully promote a crowdfunding campaign. I also had "rewards" to offer contributors. I've got my own high-value [Capstone Rails Tutorials](https://tutorials.railsapps.org/) that I offered. Plus I asked colleagues to donate free subscriptions to various screencasts and tutorials. So contributors to the campaign were actually buying tutorials. It's possible to offer rewards such as t-shirts or other swag but then you're really in the business of selling dry goods.

In my experience, a crowdfunding campaign can be used as a one-time event to raise funds to add a feature to a well-established project, if the project owner can offer high-value rewards, and can reach a large base of supporters through email lists. I don't think crowdfunding can provide ongoing support for a project.

**Comment by lovelydinosaur** [comment]
As well as the sustainability aspect, another problem with CrowdFunding is that it requires upfront design & product decisions, rather than an iterative approach. That also increases the risk on both sides (may over-commit) vs a pay-per-time relationship, which is founded on trust in the ongoing development process, and re-inforced over time.

**Comment by aaugustin** [comment]
@ximbled Russell pinged me on Twitter to ask my opinion. I wrote it. I didn't read previous comments.

The reason why you engaged in a point-by-point rebuttal of my personal experience escapes me.

Let me withdraw from this thread and please don't @-highlight me.

Thanks and sorry for having opinions.

**Comment by ximbled** [comment]
Open Source Community works when it works, if anyone feels personal insight is so valuable they can derail a discussion without having followed it to the point at which they interject. I'm sorry that I as a participant in the project haven't heard of you. But for me to rebut a comment in a discussion that I was following in the semantic order of a conversation and someone get offended by that. Here's a good example as to why Open Source will stall when it comes to securing viable funding.

**Comment by wolftune** [comment]
Snowdrift.coop's summary of [why plain threshold campaigns are lousy for FLO projects](https://snowdrift.coop/p/snowdrift/w/en/threshold-systems)

**Comment by cuducos** [comment]
+1 for a recurring model & for @andrewgodwin's brainstorm on benefits to be offered : )

**Comment by wolftune** [comment]
@cuducos Snowdrift.coop is a recurring model and @andrewgodwin we at Snowdrift.coop are already working on everything you mention, including the concerns about Patreon emphasizing proprietary special access (see our remarks at https://snowdrift.coop/p/snowdrift/w/en/othercrowdfunding#subscription ). Incidentally, we started before Patreon did, but we're still not launched because we're following ethical principles and sticking to 100% free/libre/open and not VC-backed stuff (thus we have funding challenges ourselves). _and_ our model is actually far more innovative and thus also more involved to build and make it happen.

**Comment by ghost** [comment]
Hi, last time I checked, those are USA-Only.

For Collaborating they are world wide, but if you want to publish a project you must be on USA to be able to get pay, or at least not available everywhere on the world.
